presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -837,6 +845,93 @@ public PlanNode visitSemiJoin(SemiJoinNode node, RewriteContext<Expression> cont
             return output;
         }
 
+        private PlanNode visitFilteringSemiJoin(SemiJoinNode node, RewriteContext<Expression> context)
+        {
+            Expression inheritedPredicate = context.get();
```

Comment:
I do not like `inheritedPredicate`, maybe just `predicate`?

Reply:
I do not like the `predicate`. There are many "predicates" involved in the method and some designator is nice-to-have. Also what we get from `context.get()` is called `inheritedPredicate` all over the class.